### PR TITLE
doc: update add_board_support.md

### PR DESCRIPTION
**Pull request by kaspar030**
This updates the board support doc. Could be more exhaustive but at least it is not horribly outdated anymore.

### CODE REVIEW

File: doc/add_board_support.md [hunk 1]
```diff
@@ -1,30 +1,92 @@
...
 
-        new_board = { optional=true, path="new_board" }
+- ensure [embassy architecture](#adding-an-embassy-architecture) is supported
+- in laze-project.yml:
+  - add "context" for the mcu (if it doesn't exist, yet)
+    - parent: the closest embassy arch
+      - selects: a [rustc-target](#adding-a-rustc-target-module) module
+      - if the architecture does not have a dedicated SWI, choose one now
+        and set CONFIG_SWI
+      - if the mcu is supported by probe-rs, add PROBE_RS_CHIP and PROBE_RS_PROTOCOL
```
Comment (elenaf9) [inline doc/add_board_support.md:20]: What if it isn't supported by probe-rs?

File: doc/add_board_support.md [hunk 2]
```diff
@@ -1,30 +1,92 @@
...
-1. choose a suitable base board (nrf52dk for nrf52832 based board, nrf52840dk
-   for nrf52840)
-1. copy `src/riot-rs-boards/<base board>` to `src/riot-rs-boards/<board>`
-1. replace all base board names in string literals with the new board name
-1. add the new board to the features at src/riot-rs-boards/Cargo.toml, e.g.,
+The more similar a board is to one that is already supported, the easier.
+It is usually best to copy & adapt an existing one.
 
-        new_board = { optional=true, path="new_board" }
+- ensure [embassy architecture](#adding-an-embassy-architecture) is supported
```
Comment (elenaf9) [inline doc/add_board_support.md:13]:
```suggestion
- ensure that the architecture [is supported in riot-rs-embassy](#adding-an-embassy-architecture)
```

File: doc/add_board_support.md [hunk 3]
```diff
@@ -1,30 +1,92 @@
...
+```
+
+Important are the variable RUSTC_TARGET and CARGO_TARGET_PREFIX.
+Add `--cfg foo` as needed.
+
+Chances are that if you need to add this, you'll also have to add support for the architecture to `riot-rs-threads`.
+
+## Adding an embassy architecture
+
+RIOT-rs currently supports nrf, stm32, rp and the externally (from embassy) maintained esp-rs.
```
Comment (elenaf9) [inline doc/add_board_support.md:85]: That sentence is a bit confusing. Do you mean embassy supports it?

Comment (kaspar030) [reply doc/add_board_support.md]: I tried to clarify this

File: doc/add_board_support.md (hunk 1 above)
Comment (kaspar030) [reply doc/add_board_support.md:20]: hm. esp based boards would inherit use of "espflash".

Comment (kaspar030) [reply doc/add_board_support.md]: added some more choices

**Comment by kaspar030** [comment]
Thanks @ROMemories and @elenaf9. I've committed and squashed all suggestions.

**Review by elenaf9 — APPROVED** [review]

**Pull request merged**